### DISCUSSION TITLE
[ext:joomla] - Support for intro and/or full article image?

### DISCUSSION THREAD

**Discussion by woluweb** [joomlatools/joomlatools-pages#673] [Q&A] [answered]
Pages fetches the Article image with `<?= $article->image->url; ?>` But what if there is an Intro Image *and* a Full Article Image?
Is it possible to fetch each of them?

**Comment by johanjanssens** [comment, marked as answer]
At the moment Pages only support a single image through the image property. Pages checks if the article has an fulltext image and use it, if not it will check if there is an intro text image and use that. See: https://github.com/joomlatools/joomlatools-pages/blob/master/contrib/extensions/joomla/model/entity/article.php#L198

It doesn't make much sense to have 2 images for a single article. This makes it not possible to automatically choose the right image, for example Pages has out of the box support for Opengraph and it requires an image to be valid. See: https://ogp.me/

If an image is defined for a page Pages automatically adds Opengraph metadata to the page to make it socially discoverable. In case of Joomla articles or categories it needs to have a single image for that, hence only having a 'image' property and not 2.

If your article has other images those would be considered content images, what we could do however is add an additional property to give access to all the images that are defined or embedded in the text. Could be a feature worthwhile adding. I have [made a note](https://github.com/joomlatools/joomlatools-pages/issues/175) of that and will look into it.
